libpebble3/src/commonMain/kotlin/io/rebble/libpebblecommon/datalogging/Datalogging.kt

```diff
@@ -3,43 +3,344 @@ package io.rebble.libpebblecommon.datalogging
 import co.touchlab.kermit.Logger
 import io.rebble.libpebblecommon.SystemAppIDs.SYSTEM_APP_UUID
 import io.rebble.libpebblecommon.connection.WebServices
+import io.rebble.libpebblecommon.database.dao.HealthDao
+import io.rebble.libpebblecommon.database.dao.insertHealthDataWithPriority
+import io.rebble.libpebblecommon.database.entity.HealthDataEntity
+import io.rebble.libpebblecommon.database.entity.OverlayDataEntity
 import io.rebble.libpebblecommon.di.LibPebbleCoroutineScope
+import io.rebble.libpebblecommon.health.OverlayType
 import io.rebble.libpebblecommon.services.WatchInfo
 import io.rebble.libpebblecommon.structmapper.SBytes
 import io.rebble.libpebblecommon.structmapper.SUInt
 import io.rebble.libpebblecommon.structmapper.StructMappable
 import io.rebble.libpebblecommon.util.DataBuffer
 import io.rebble.libpebblecommon.util.Endian
-import kotlinx.coroutines.launch
 import kotlin.uuid.Uuid
+import kotlinx.coroutines.launch
 
 class Datalogging(
-    private val libPebbleCoroutineScope: LibPebbleCoroutineScope,
-    private val webServices: WebServices,
+        private val libPebbleCoroutineScope: LibPebbleCoroutineScope,
+        private val webServices: WebServices,
+        private val healthDao: HealthDao,
+        private val knownWatchDao: io.rebble.libpebblecommon.database.dao.KnownWatchDao,
 ) {
     private val logger = Logger.withTag("Datalogging")
 
-    fun logData(uuid: Uuid, tag: UInt, data: ByteArray, watchInfo: WatchInfo) {
+    fun logData(uuid: Uuid, tag: UInt, data: ByteArray, watchInfo: WatchInfo, itemSize: UShort) {
         if (uuid == SYSTEM_APP_UUID) {
-            if (tag == MEMFAULT_CHUNKS_TAG) {
-                libPebbleCoroutineScope.launch {
-                    val chunk = MemfaultChunk()
-                    chunk.fromBytes(DataBuffer(data.toUByteArray()))
-                    val chunkBytes = chunk.bytes.get()
-                    webServices.uploadMemfaultChunk(chunkBytes.toByteArray(), watchInfo)
+            when (tag) {
+                MEMFAULT_CHUNKS_TAG -> {
+                    libPebbleCoroutineScope.launch {
+                        val chunk = MemfaultChunk()
+                        chunk.fromBytes(DataBuffer(data.toUByteArray()))
+                        val chunkBytes = chunk.bytes.get()
+                        webServices.uploadMemfaultChunk(chunkBytes.toByteArray(), watchInfo)
+                    }
+                }
+                HEALTH_STEPS_TAG -> {
+                    libPebbleCoroutineScope.launch {
```

Comment:
Probably should move the parsing code out of this class into a health-specific class